### PR TITLE
test: Add MTU verification tests for EVPN networks

### PR DESCRIPTION
**Pull request by meinali566**
## Testing Details

  Add E2E tests to verify MTU configuration across the EVPN fabric, ensuring large packets traverse between pods on different nodes without fragmentation.

  The test validates per IP family (IPv4/IPv6):
  1. MTU boundary relationships on every node:
     podMTU <= vxlanMTU <= (underlayMTU - vxlanOverhead)
     VXLAN overhead: 50 bytes for IPv4, 70 bytes for IPv6
  2. Packet delivery at MTU boundary using Don't Fragment (DF) bit:
     - Packets at the MTU limit succeed (no silent fragmentation)
     - Packets exceeding the MTU limit fail

  ## Test Coverage

  The test runs for all EVPN network types:
  - ✅ Layer 3 CUDN EVPN IP-VRF
  - ✅ Layer 2 CUDN EVPN MAC-VRF
  - ✅ Layer 2 CUDN EVPN MAC-VRF and IP-VRF

## Summary by CodeRabbit

* **New Features**
  * EVPN support: node and pod controllers to manage VTEPs, VXLAN/SVI/OVS port lifecycle and deterministic device naming; per-pod neighbor/FDB programming and cleanup.
* **Tests**
  * Extensive unit/integration suites and new end-to-end MTU/connectivity validations (table-driven MTU checks, payload/DF verification).
* **Documentation**
  * Tighter unmanaged VTEP discovery rules to exclude VIPs and secondary addresses.
* **Chores**
  * Bumped FRR demo image, added FRR readiness gating, and granted node RBAC read access to VTEP resources.

### CODE REVIEW

File: test/e2e/route_advertisements.go [hunk 1]
```diff
@@ -2720,6 +2721,166 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+
+					ginkgo.By("Verify pod interface MTU")
+					podIfaceOutput, err := e2epodoutput.RunHostCmd(testPod[0].Namespace, testPod[0].Name, "ip -j link show ovn-udn1")
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					podMTU, err := getMTUByInterfaceName(podIfaceOutput, "ovn-udn1")
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					framework.Logf("Pod interface MTU: %d", podMTU)
+					// Verify pod MTU is reasonable for overlay networking
+					gomega.Expect(podMTU).To(gomega.BeNumerically(">=", 1400))
```
Comment (jechen0648) [inline test/e2e/route_advertisements.go]: @anuragthehatter is it always greater than 1400?

File: test/e2e/route_advertisements.go [hunk 2]
```diff
@@ -2720,6 +2721,166 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+						if len(vxlanDevices) > 0 {
+							vxlanName := vxlanDevices[0].Name
+							vxlanLinkOutput, err := infraprovider.Get().ExecK8NodeCommand(node.Name, []string{"ip", "-j", "link", "show", vxlanName})
+							gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+							vxlanMTU, err := getMTUByInterfaceName(vxlanLinkOutput, vxlanName)
+							gomega.Expect(err).NotTo(gomega.HaveOccurred())
+							framework.Logf("Node %s VXLAN device %s MTU: %d", node.Name, vxlanName, vxlanMTU)
+
+							// Verify VXLAN MTU accounts for VXLAN overhead (typically 50 bytes)
```
Comment (jechen0648) [inline test/e2e/route_advertisements.go]: assume we get podMTU of pod, and nodeMTU on eth0 of the node, vxlanMTU of the vxlan interface used in vxlan tunnel
and we know vxlanOverheadRoom ( 50 bytes for IPv4, 70 bytes for IPv6)
we want to verify:
podMTU   <=  vxlanMTU   <=  (nodeMTU - vxlanOverheadRoom)

from your log:
  I0319 08:43:18.136863 53334 route_advertisements.go:2819] Node ovn-control-plane VXLAN device vxbgp4139 MTU: 1500
  I0319 08:43:18.251453 53334 route_advertisements.go:2833] Node ovn-control-plane underlay interface breth0 MTU: 1500
  I0319 08:43:18.455723 53334 route_advertisements.go:2819] Node ovn-worker VXLAN device vxbgp4139 MTU: 1500
  I0319 08:43:18.542949 53334 route_advertisements.go:2833] Node ovn-worker underlay interface breth0 MTU: 1500
  I0319 08:43:18.730716 53334 route_advertisements.go:2819] Node ovn-worker2 VXLAN device vxbgp4139 MTU: 1500
  I0319 08:43:18.994687 53334 route_advertisements.go:2833] Node ovn-worker2 underlay interface breth0 MTU: 1500

it does not look right

File: test/e2e/route_advertisements.go [hunk 3]
```diff
@@ -2720,6 +2721,166 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+						testPod[1].Name,
+						testNetworkName,
+						family,
+					)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					gomega.Expect(otherPodIP).NotTo(gomega.BeEmpty())
+
+					testPacketSizes := []int{
+						1400, // Normal size, should always work
+						1450, // Near MTU limit
```
Comment (jechen0648) [inline test/e2e/route_advertisements.go]: Should test by sending packet below limit (should pass) as well as sending packet over limit (should fail)

Comment (meinali566) [reply test/e2e/route_advertisements.go]: I've added `over limit` and `boundary limit` MTU check

File: test/e2e/route_advertisements.go (hunk 2 above)
Comment (meinali566) [reply test/e2e/route_advertisements.go]: thanks. I updated code first but this bug shouldn't be fixed. I will check test results after fix merged

File: test/e2e/route_advertisements.go [hunk 4]
```diff
@@ -2720,6 +2721,166 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+						f.ClientSet,
+						testNamespace.Name,
+						testNamespace.Name+"-netexec-pod",
+						func(p *corev1.Pod) {
+							p.Spec.Containers[0].Args = []string{"netexec"}
+							p.Spec.NodeName = nodes.Items[i].Name
+							p.Labels = map[string]string{"app": "netexec-pod"}
+						},
+					)
+				}
```
[the remark this answers is not in the document]
Comment (meinali566) [reply test/e2e/route_advertisements.go:2755]: thanks, resolved

File: test/e2e/route_advertisements.go [hunk 5]
```diff
@@ -2720,6 +2721,166 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+						underlayMTU, err := getMTUByInterfaceName(underlayLinkOutput, underlayIfName)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						framework.Logf("Node %s underlay interface %s MTU: %d", node.Name, underlayIfName, underlayMTU)
+
+						// Verify underlay has sufficient headroom for overlay
+						// VXLAN overhead = 50 bytes (8 VXLAN + 8 UDP + 20 IPv4 + 14 Ethernet)
+						minUnderlayMTU := 1500
+						gomega.Expect(underlayMTU).To(gomega.BeNumerically(">=", minUnderlayMTU))
```
[the remark this answers is not in the document]
Comment (meinali566) [reply test/e2e/route_advertisements.go]: resolved

File: test/e2e/route_advertisements.go (hunk 1 above)
Comment (meinali566) [reply test/e2e/route_advertisements.go]: @jechen0648 I think you're right. it shouldn't be always greater then 1400. I delete those comparison, just keep headroom comparison.

Comment (jechen0648) [reply test/e2e/route_advertisements.go]: maybe you want to correct your PR description, as podMTU >= 1400 is still mentioned there

File: test/e2e/route_advertisements.go [hunk 6]
```diff
@@ -2720,6 +2700,177 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+						e2eskipper.Skipf("IP family %v not supported", family)
+					}
+
+					// Calculate VXLAN overhead based on IP family
+					// IPv4: 8 (VXLAN) + 8 (UDP) + 20 (IPv4) + 14 (Ethernet) = 50 bytes
+					// IPv6: 8 (VXLAN) + 8 (UDP) + 40 (IPv6) + 14 (Ethernet) = 70 bytes
+					vxlanOverhead := 50
+					if family == utilnet.IPv6 {
+						vxlanOverhead = 70
+					}
```
[the remark this answers is not in the document]
Comment (meinali566) [reply test/e2e/route_advertisements.go:2778]: @coderabbitai but it cannot check ipv6 edge value, right?

File: test/e2e/route_advertisements.go [hunk 7]
```diff
@@ -1940,9 +1941,16 @@ var _ = ginkgo.Describe("BGP: For BGP configured networks", feature.RouteAdverti
...
+			// TODO: Only unmanaged mode is supported currently
+			// TODO: Enable ipv6 once FRR supports it
+			// Derive VTEP subnet from the kind network CIDRs(unmanaged mode)
+			// NOTE: FRR does not support anything else than IPv4 vteps today
+			kindNetwork, err := infraprovider.Get().GetNetwork("kind")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			kindV4Subnet, _, err := kindNetwork.IPv4IPv6Subnets()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			vtepSubnets := []string{kindV4Subnet}
+			framework.Logf("Networks used for EVPN VTEPs: %v", vtepSubnets)
```
[the remark this answers is not in the document]
Comment (meinali566) [reply test/e2e/route_advertisements.go:1953]: @coderabbitai we should keep current design for future vtep support

**Comment by jechen0648** [comment]
Because  OVN/OVS don’t support doing any fragmentation so expecting the pod to fragment and then OVN to unfragment and fragment again is not going to work.  Close this PR, a different e2e will be added to test with TCP MSS clamping over EVPN

/close
